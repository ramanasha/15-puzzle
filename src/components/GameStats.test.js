import renderer from 'react-test-renderer';
import React from 'react';
import GameStats from './GameStats';
import * as GameStates from '../lib/game-status';

describe('GameStats', () => {
  it('should render correctly given a GAME_OVER state', () => {
    const tree = renderer
      .create(
        <GameStats moves={30} seconds={120} gameState={GameStates.GAME_OVER} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly given a GAME_STARTED state', () => {
    const tree = renderer
      .create(
        <GameStats moves={0} seconds={1} gameState={GameStates.GAME_STARTED} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly given a GAME_IDLE state', () => {
    const tree = renderer
      .create(
        <GameStats moves={12} seconds={60} gameState={GameStates.GAME_IDLE} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
