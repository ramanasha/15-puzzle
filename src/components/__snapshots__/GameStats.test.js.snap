exports[`GameStats should render correctly given a GAME_IDLE state 1`] = `
<div
  className="gPhJsb">
  <div
    className="fyeoun">
    Moves counter: 
    12
  </div>
  <div
    className="fyeoun">
    Time elapsed: 
    60
    s
  </div>
</div>
`;

exports[`GameStats should render correctly given a GAME_OVER state 1`] = `
<div
  className="gPhJsb">
  <div
    className="fyeoun">
    Moves counter: 
    30
  </div>
  <div
    className="fyeoun">
    Time elapsed: 
    120
    s
  </div>
  <div
    className="gameStateMessage">
    Congrats!
  </div>
</div>
`;

exports[`GameStats should render correctly given a GAME_STARTED state 1`] = `
<div
  className="gPhJsb">
  <div
    className="fyeoun">
    Moves counter: 
    0
  </div>
  <div
    className="fyeoun">
    Time elapsed: 
    1
    s
  </div>
</div>
`;
